src/Analyser/MutatingScope.php

```diff
@@ -503,6 +503,16 @@ public function hasVariableType(string $variableName): TrinaryLogic
 	/** @api */
 	public function getVariableType(string $variableName): Type
 	{
+		if (!$this->treatPhpDocTypesAsCertain) {
+			if ($variableName === 'this' && $this->hasVariableType('this')->yes()) {
+				return $this->variableTypes['this']->getType();
+			}
+			if (!array_key_exists(sprintf('$%s', $variableName), $this->nativeExpressionTypes)) {
+				return new MixedType();
+			}
+			return $this->nativeExpressionTypes[sprintf('$%s', $variableName)];
```

Comment:
A slightly more expressive approach that also removes the need to `sprintf()` twice:

```suggestion
			return $this->nativeExpressionTypes[sprintf('$%s', $variableName)] ?? new MixedType();
```

Reply:
Thank you for your suggestions!